templates/hosting/init.js

```diff
@@ -1,5 +1,30 @@
 if (typeof firebase === 'undefined') throw new Error('hosting/init-error: Firebase SDK not detected. You must include it before /__/firebase/init.js');
+function hasFirebaseFeature(feature) {
+  return firebase && typeof firebase[feature] === "function";
+}
 /*--CONFIG--*/
+/*--EMULATORCONFIG--*/
 if (firebaseConfig) {
+  // Database has to be configured *before* initializeApp, while others can be configured after.
+  if (emulatorConfig.database) {
+    console.log("Configuring Database SDK to point at emulator: " + emulatorConfig.database);
+    firebaseConfig.databaseURL = emulatorConfig.database
+  }
+
   firebase.initializeApp(firebaseConfig);
-}
+
+  if (emulatorConfig) {
+    if (hasFirebaseFeature("firestore") && emulatorConfig.firestore) {
+      console.log("Configuring Firestore SDK to point at emulator: " + emulatorConfig.firestore);
+      firebase.firestore().settings({
```

Comment:
Unfortunately, this only gives you a `firestore.Database` instance with the right settings (which is immediately discarded since we don't store the reference). It won't have any side effect like changing the default settings etc., so existing code in the web app like `firebase.firestore().xxx` still connect to production.